src/parser.ts

```diff
@@ -151,11 +151,79 @@ export class Parser {
         return meta;
     }
 
+    private preProcessTemplateBody(templateBody: string) {
+        /**
+         * This function is supposed to do the following preprocessing.
+         * 1. Wrap the value of template_title and template_tags with double quotes if it isn't already.
+         */
+
+        templateBody = templateBody.trimStart();
+
+        const getVariableDefinitionsBlock = () => {
+            const allLines = templateBody.split("\n");
+
+            if (allLines.length === 0 || allLines[0] !== "---") {
+                return null;
+            }
+
+            const allMatchesAfterFirstMatch = allLines.map((val, index) => {
+                if (index && val === "---") {
+                    return index;
+                } else {
+                    return null;
+                }
+            }).filter(val => !!val);
+
+            if (!allMatchesAfterFirstMatch.length) {
+                return null;
+            }
+
+            const blockEndIndex = allMatchesAfterFirstMatch[0];
+            return allLines.slice(0, blockEndIndex + 1).join("\n");
+        }
+
+        const wrapInQuotes = (definitionsBlock: string, properties: string[]) => {
+            for (const prop of properties) {
+                const pattern = new RegExp(`^ *${prop}:.*`, "gm");
```

Comment:
You should use `[^\S\n]` rather than a literal space here. It will allow you to catch all (non-newline) whitespace.

Reply:
done